File: src/main/java/com/github/kuro46/embedscript/script/processor/AbstractExecutor.java
package com.github.kuro46.embedscript.script.processor;

import com.google.common.collect.ImmutableList;
import org.bukkit.entity.Player;

public abstract class AbstractExecutor implements Processor.Executor {
    @Override
    public boolean check(Player trigger, ImmutableList<String> matchedValues) {
        return true;
    }

    @Override
    public void prepareExecute(Player trigger, ImmutableList<String> matchedValues) {

    }

    @Override
    public void beginExecute(Player trigger, ImmutableList<String> matchedValues) {

    }

    @Override
    public void endExecute(Player trigger, ImmutableList<String> matchedValues) {

    }
}

File: src/main/kotlin/com/github/kuro46/embedscript/script/processor/AbstractExecutor.kt
package com.github.kuro46.embedscript.script.processor

import com.google.common.collect.ImmutableList
import org.bukkit.entity.Player

abstract class AbstractExecutor : Processor.Executor {
    override fun check(trigger: Player, matchedValues: ImmutableList<String>): Boolean {
        return true
    }

    override fun prepareExecute(trigger: Player, matchedValues: ImmutableList<String>) {

    }

    override fun beginExecute(trigger: Player, matchedValues: ImmutableList<String>) {

    }

    override fun endExecute(trigger: Player, matchedValues: ImmutableList<String>) {

    }
}
